### PR TITLE
enh(ci): add platform deploy from nightly

### PR DESCRIPTION
**Pull request by tuntoja**
## Description

Add a deploy of nightly rpms to a platform

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/web.yml [hunk 1]
```diff
@@ -683,3 +683,14 @@ jobs:
           major_version: ${{ needs.get-version.outputs.major_version }}
           minor_version: ${{ needs.get-version.outputs.minor_version }}
           stability: ${{ needs.get-version.outputs.stability }}
+
+  deploy-platform-nightly:
+    needs: [deliver-rpm]
+    if: github.event.schedule == '0 3 * * *'
```
Comment (bdauria) [inline .github/workflows/web.yml:689]: Use "${{ github.event_name }}" = "schedule"  to stay consistent with the get-version condition

File: .github/workflows/web.yml [hunk 2]
```diff
@@ -714,3 +714,27 @@ jobs:
...
+          bu: RD
+          disk_size: 30
+          poller_number: 1
+          central_instance: t3a.medium
+          poller_instance: t3a.medium
+          system_name: Alma8
+          centreon_branch: develop
+          configured_resources: false
+          install_business_modules: false
+          instance_password: Centreon!2021
```
Comment (bdauria) [inline .github/workflows/web.yml:739]: should be a secret

File: .github/workflows/web.yml [hunk 3]
```diff
@@ -683,3 +683,14 @@ jobs:
...
+  deploy-platform-nightly:
+    needs: [deliver-rpm]
+    if: github.event.schedule == '0 3 * * *'
+    runs-on: [common]
+    steps:
+      - name: Trigger platform deployment
+        uses: ./.github/actions/nightly-platform-deploy
+        with:
+          jambon: b
+          beurre: a
```
[the remark this answers is not in the document]
Comment (tuntoja) [reply .github/workflows/web.yml]: yes

File: .github/workflows/web.yml [hunk 4]
```diff
@@ -794,3 +794,33 @@ jobs:
...
+    # needs:
+    #   [
+    #     get-version,
+    #     api-integration-test,
+    #     e2e-test,
+    #     performances-test,
+    #     legacy-e2e-test,
+    #     deliver-rpm
+    #   ]
+    # if: ${{ github.event_name == 'schedule' && contains(fromJson('["unstable"]'), needs.get-version.outputs.stability) }}
```
Comment (sc979) [inline .github/workflows/web.yml:809]:
```suggestion
```

Comment (sc979) [reply .github/workflows/web.yml:809]: change him

**Review by sc979 — APPROVED** [review]

File: .github/actions/nightly-platform-deploy/action.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
+name: "nightly-platform-deploy"
+description: "Deploy a platform with packages from nightly builds."
+inputs:
+  deploy_type:
+    description: "The type of platform deployment (install or update)."
+    required: false
+  ref:
+    description: "Reference branch for cod."
```
Comment (bdauria) [inline .github/actions/nightly-platform-deploy/action.yml:8]:
```suggestion
    description: "Reference branch for centreon on demand"
```

File: .github/actions/nightly-platform-deploy/action.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
+name: "nightly-platform-deploy"
+description: "Deploy a platform with packages from nightly builds."
+inputs:
+  deploy_type:
+    description: "The type of platform deployment (install or update)."
```
Comment (bdauria) [inline .github/actions/nightly-platform-deploy/action.yml:5]:
```suggestion
    description: "The type of platform deployment (install or update)"
```

File: .github/actions/nightly-platform-deploy/action.yml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    required: false
+  ref:
+    description: "Reference branch for cod."
+    required: true
+  bu:
+    description: "Business unit to use."
```
Comment (bdauria) [inline .github/actions/nightly-platform-deploy/action.yml:11]:
```suggestion
    description: "Business unit to use"
```

File: .github/actions/nightly-platform-deploy/action.yml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+          # Deploy a new fresh instance if destroy suceeded
+          echo "[INFO]: Deploying nightly COD instance with latest ${{ inputs.centreon_branch }} state."
+          curl --fail \
+          -s \
+          -o /dev/null \
+          -X POST \
+          -H "Authorization: Bearer ${{inputs.cod_service_token}}" \
+          -H "Accept: application/vnd.github.v3+json" \
+          https://api.github.com/repos/centreon/centreon-on-demand/actions/workflows/deploy.yml/dispatches \
+          -d '{"ref": "main", "inputs":{"bu":"${{ inputs.bu }}", "deployment_profile":"Standard profile", "poller_number":"${{ inputs.poller_number }}", "system_name":"${{ inputs.system_name }}", "centreon_branch":"${{ inputs.centreon_branch }}", "custom_centreon_web_admin_password":"${{ inputs.instance_password }}", "configured_resources":"${{ inputs.configured_resources }}", "install_business_modules":"${{ inputs.install_business_modules }}" }}'
```
Comment (kduret) [inline .github/actions/nightly-platform-deploy/action.yml:69]:
```suggestion
            -s \
            -o /dev/null \
            -X POST \
            -H "Authorization: Bearer ${{inputs.cod_service_token}}" \
            -H "Accept: application/vnd.github.v3+json" \
            https://api.github.com/repos/centreon/centreon-on-demand/actions/workflows/deploy.yml/dispatches \
            -d '{"ref": "main", "inputs":{"bu":"${{ inputs.bu }}", "deployment_profile":"Standard profile", "poller_number":"${{ inputs.poller_number }}", "system_name":"${{ inputs.system_name }}", "centreon_branch":"${{ inputs.centreon_branch }}", "custom_centreon_web_admin_password":"${{ inputs.instance_password }}", "configured_resources":"${{ inputs.configured_resources }}", "install_business_modules":"${{ inputs.install_business_modules }}" }}'
```

**Review by kduret — COMMENTED** [review]
not easier to use github cli ? https://cli.github.com/manual/gh_workflow_run
do not apply on this PR

**Review by kduret — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**